src/proto/streams/streams.rs

```diff
@@ -402,44 +404,61 @@ where
         let id = frame.stream_id();
         let promised_id = frame.promised_id();
 
-        let res = {
-            let stream = match me.store.find_mut(&id) {
-                Some(stream) => stream.key(),
-                None => return Err(RecvError::Connection(Reason::PROTOCOL_ERROR)),
-            };
+        // First, ensure that the initiating stream is still in a valid state.
+        let parent_key = match me.store.find_mut(&id) {
+            Some(stream) => {
+                if id > me.actions.recv.max_stream_id() {
```

Comment:
It took me a minute to go look this up to remember why the check was done. What do you think of a tiny comment hear explaining that if a `GOAWAY` has been started, frames higher than the sent id can be ignored?